chainer/optimizers/adam.py

```diff
@@ -210,6 +278,18 @@ def lr(self):
             DeprecationWarning)
         return self.alpha_t
 
+    @property
+    def bounds(self):
+        if self.t == 0:
+            raise RuntimeError(
+                'Can\'t determine the bounds of AdaBound optimizer '
+                'because the update steps have not been started.')
+        hp = self.hyperparam
+        final_lr = hp.final_lr * hp.alpha / self.state['initial_alpha']
```

Comment:
Could you add a comment explaining that `hp.alpha / self.state['initial_alpha']` is a workaround to enable some of `chaienr.training.extensions` that change their target optimizer's parameters during training?